app/src/main/java/swati4star/createpdf/adapter/ViewFilesAdapter.java

```diff
@@ -324,6 +341,67 @@ public String string(@StringRes int resId) {
             ButterKnife.bind(this, itemView);
         }
     }
+    private void setPassword(final String fileName) {
+        final MaterialDialog dialog = new MaterialDialog.Builder(mActivity)
+                .title(R.string.set_password)
+                .customView(R.layout.custom_dialog, true)
+                .positiveText(android.R.string.ok)
+                .negativeText(android.R.string.cancel)
+                .build();
+
+        mPositiveAction = dialog.getActionButton(DialogAction.POSITIVE);
+        mPasswordInput = dialog.getCustomView().findViewById(R.id.password);
+        mPasswordInput.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        mPositiveAction.setEnabled(s.toString().trim().length() > 0);
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable input) {
+                        if (StringUtils.isEmpty(input)) {
+                            Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                                    R.string.snackbar_password_cannot_be_blank,
+                                    Snackbar.LENGTH_LONG).show();
+                        } else {
+                            mPassword = input.toString();
+
+                        }
+                    }
+                });
+        dialog.show();
+        mPositiveAction.setEnabled(false);
+        mPositiveAction.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                try {
+                    doEncryption(fileName);
+                } catch (IOException | DocumentException e) {
+                    e.printStackTrace();
+                }
+                dialog.dismiss();
+                notifyDataSetChanged();
+                Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                        R.string.password_added,
+                        Snackbar.LENGTH_LONG).show();
+            }
+        });
+
+    }
+    private void doEncryption(String path) throws IOException, DocumentException {
+        PdfReader reader = new PdfReader(path);
+        PdfStamper stamper = new PdfStamper(reader,new FileOutputStream("/storage/emulated/0/PDFfiles/sample.pdf"));
```

Comment:
Why are you using this constant path

Reply:
If I delete the file, then how the reader and stamper are gonna encrypt the file.. i guess then it is same as creating another pdf with encryption.

Reply:
I am creating an OutputFile from the file to be encrypted.

Reply:
Then, do this:
Suppose the user wants to encrypt `abc.pdf` file, then create the encrypted version of the file with name `abc_encrypted.pdf`.
Now, the user will have both. He can anyway rename the file 👍

Reply:
Okay, so the PDF View is gonna display both encrypted as well as normal one, or should I delete the non encrypted one ?

Reply:
Don't delete. Let's keep both.

Reply:
Okay Sure